src/Swashbuckle.AspNetCore.SwaggerGen/SwaggerGenerator/SwaggerGenerator.cs

```diff
@@ -58,14 +59,25 @@ public async Task<OpenApiDocument> GetSwaggerAsync(string documentName, string h
                 filter.Apply(swaggerDoc, filterContext);
             }
 
+            foreach (var filter in _options.DocumentAsyncFilters)
+            {
+                await filter.Apply(swaggerDoc, filterContext);
+            }
+
             swaggerDoc.Components.Schemas = new SortedDictionary<string, OpenApiSchema>(swaggerDoc.Components.Schemas, _options.SchemaComparer);
 
             return swaggerDoc;
         }
 
         public OpenApiDocument GetSwagger(string documentName, string host = null, string basePath = null)
         {
-            var (applicableApiDescriptions, swaggerDoc, schemaRepository) = GetSwaggerDocumentWithoutFilters(documentName, host, basePath);
+            if (_options.HasAsyncFilters)
```

Comment:
I don't think there's a need to do this - if someone needs a filter that's asynchronous and has others that don't need it, they should be able to implement it with both interfaces, not force them to refactor all their others to make everything match.

Reply:
I don't think that is what this code will do, at least what I meant that it does is that it fails if you use an async filter with the sync `GetSwagger()`.

I am not 100% sure when `GetSwagger()` is called instead of `GetSwaggerAsync()` the middleware seems to prefer the async version of the sync version.

Reply:
When I have time to clone the code locally and dig through it in an IDE I can take a better look at what's the best way to go about this to try and make it "just work" for the async case.

If we were to make breaking changes in the future we'd probably remove the sync versions if we also have async.

Reply:
I have now removed this check (and the corresponding property on SwaggerGeneratorOptions)